storage/pebble.go

```diff
@@ -172,17 +174,16 @@ func (s *Pebble) BlockIterator(fromBlockNum, toBlockNum uint64) (Iterator[*types
 		return nil, multierr.Append(snap.Close(), err)
 	}
 
-	return &PebbleBlockIter{i: it, s: snap}, nil
+	return &PebbleBlockIter{i: it, s: snap, ascending: ascending}, nil
 }
 
 func (s *Pebble) TxIterator(
```

Comment:
instead of sending a flag for reversing the iterator, it might be less convoluted to create a method for the functionality you want to have, in that case, `ReverseTxIterator`

Doing that you don't need to tweak ranges outside the iterator logic itself.